src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs

```diff
@@ -41,7 +39,7 @@ public ConditionalAttributeDiscoverer(IMessageSink diagnosticMessageSink)
                 .Discover(discoveryOptions, testMethod, attributeInfo)
                 .Select(testCase => new SkipReasonTestCase
                 {
-                    IsTheory = isTheory,
+                    IsTheory = testCase.GetType() == typeof(XunitTheoryTestCase),
```

Comment:
Eh, I haven't seen it in our repos but it's a supported scenario from XUnit. Using `IsAssignableFrom` here would just be for sake of correctness in-case we happen to do it in the future.